text/0118-search-api.md

```diff
@@ -912,6 +913,20 @@ The documents containing ALL the query words (i.e. in the `q` parameter) are ret
 
 Only the documents containing ALL the query words (i.e. in the `q` parameter) are returned by Meilisearch. If Meilisearch doesn't have enough documents to fit the requested `limit`, it returns the documents found without trying to match more documents.
 
+#### 3.1.17. `attributesToSearchOn`
+
+- Type: Array of String (POST) | String (GET)
+- Required: False
+- Default: `null`
+
+Defines which `searchableAttributes` the query will search on.
+
+- If `attributesToSearchOn` is not set or set to `null`, then the query will search on all `searchableAttributes`.
+- Sending the attributes in a different order than the order set in the settings `searchableAttributes` doesn't reorder the fields' rank for the `Attributes` ranking rule
+- 🔴 Sending a value with a different type than `Array of String`(POST), `String`(GET) or `null` for `attributesToSearchOn` returns an [invalid_attributes_to_search_on](0061-error-format-and-definitions.md#invalid_search_attributes_to_search_on) error.
```

Comment:
What happens if you send an empty array?

Reply:
Empty result

Reply:
Should that be specified or is it common practice in the API?

Reply:
I think that defining an empty attributesToSearchOn list means that you search in no attributes, which implies no results. Do you see any other behavior?

Reply:
I guess not, I was just wondering if it might be surprising for a user to leave it empty and then be confused about getting no results

Reply:
I'm more on the side of the error if the array is sent with an empty form, so the developer can't break the search and is aware of it; By doing this, we can catch it before they have to understand what's wrong and fix it; I see no point in returning 0 results, am I wrong?

Reply:
Maybe we could return the hits as if the query was empty?

Reply:
> I'm more on the side of the error if the array is sent with an empty form, so the developer can't break the search and is aware of it; By doing this, we can catch it before they have to understand what's wrong and fix it; I see no point in returning 0 results, am I wrong?

I agree about the fact that would be useless to set `[]`, however, it's a logical behavior from Meilisearch in my sense. Moreover, it's consistent with the [`searchableAttributes` that accepts an empty array](https://github.com/meilisearch/specifications/blob/main/text/0123-searchable-attributes-setting-api.md#3321-request-payload-definition)

> Maybe we could return the hits as if the query was empty?

It doesn't seems logical to me to replace the search with a placeholder one because `attributeToSearchOn` is empty. 🤔

Reply:
+1 @gmourier , let's be consistent and revisit this when we're ready for a v2